x/concentrated-liquidity/client/query_proto_wrap.go

```diff
@@ -33,7 +33,7 @@ func (q Querier) UserPositions(ctx sdk.Context, req clquery.UserPositionsRequest
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	userPositions, err := q.Keeper.GetUserPositions(ctx, sdkAddr, req.PoolId)
+	userPositions, pageRes, err := q.Keeper.GetUserPositionsSerialized(ctx, sdkAddr, req.PoolId, req.Pagination)
```

Comment:
nit: right below we iterate over `userPositions` again which is redundant. We could instead achieve that in a single iteration inside `GetUserPositionsSerialized`

Reply:
good call, changed here https://github.com/osmosis-labs/osmosis/pull/5377/commits/b79937e409043d0054a5cfe0a14d21122db3c536